editor/plugins/mesh_instance_3d_editor_plugin.cpp

```diff
@@ -58,200 +58,159 @@ void MeshInstance3DEditor::edit(MeshInstance3D *p_mesh) {
 
-			Ref<ConcavePolygonShape3D> shape = mesh->create_trimesh_shape();
-			if (shape.is_null()) {
-				return;
+			if (p_verbose && shapes.is_empty()) {
+				err_dialog->set_text(TTR("Couldn't create a single collision shape."));
+				err_dialog->popup_centered();
 			}
-
-			CollisionShape3D *cshape = memnew(CollisionShape3D);
-			cshape->set_shape(shape);
-			cshape->set_transform(node->get_transform());
-
-			Node *owner = get_tree()->get_edited_scene_root();
-
-			EditorUndoRedoManager *ur = EditorUndoRedoManager::get_singleton();
-
-			ur->create_action(TTR("Create Trimesh Static Shape"));
-
-			ur->add_do_method(node->get_parent(), "add_child", cshape, true);
-			ur->add_do_method(node->get_parent(), "move_child", cshape, node->get_index() + 1);
-			ur->add_do_method(cshape, "set_owner", owner);
-			ur->add_do_method(Node3DEditor::get_singleton(), SceneStringNames::get_singleton()->_request_gizmo, cshape);
-			ur->add_do_reference(cshape);
-			ur->add_undo_method(node->get_parent(), "remove_child", cshape);
-			ur->commit_action();
 		} break;
 
-		case MENU_OPTION_CREATE_SINGLE_CONVEX_COLLISION_SHAPE:
-		case MENU_OPTION_CREATE_SIMPLIFIED_CONVEX_COLLISION_SHAPE: {
-			if (node == get_tree()->get_edited_scene_root()) {
-				err_dialog->set_text(TTR("Can't create a single convex collision shape for the scene root."));
+		case SHAPE_TYPE_SIMPLIFIED_CONVEX: {
+			shapes.push_back(p_mesh->create_convex_shape(true, true));
+
+			if (p_verbose && shapes.is_empty()) {
+				err_dialog->set_text(TTR("Couldn't create a simplified collision shape."));
 				err_dialog->popup_centered();
-				return;
 			}
+		} break;
 
-			bool simplify = (p_option == MENU_OPTION_CREATE_SIMPLIFIED_CONVEX_COLLISION_SHAPE);
+		case SHAPE_TYPE_MULTIPLE_CONVEX: {
+			Ref<MeshConvexDecompositionSettings> settings = Ref<MeshConvexDecompositionSettings>();
+			settings.instantiate();
+			settings->set_max_convex_hulls(32);
+			settings->set_max_concavity(0.001);
 
-			Ref<ConvexPolygonShape3D> shape = mesh->create_convex_shape(true, simplify);
+			shapes = p_mesh->convex_decompose(settings);
 
-			if (shape.is_null()) {
-				err_dialog->set_text(TTR("Couldn't create a single convex collision shape."));
+			if (p_verbose && shapes.is_empty()) {
+				err_dialog->set_text(TTR("Couldn't create any collision shapes."));
 				err_dialog->popup_centered();
-				return;
 			}
-			EditorUndoRedoManager *ur = EditorUndoRedoManager::get_singleton();
+		} break;
 
-			if (simplify) {
-				ur->create_action(TTR("Create Simplified Convex Shape"));
-			} else {
-				ur->create_action(TTR("Create Single Convex Shape"));
-			}
+		default:
+			break;
+	}
+	return shapes;
+}
 
-			CollisionShape3D *cshape = memnew(CollisionShape3D);
-			cshape->set_shape(shape);
-			cshape->set_transform(node->get_transform());
+void MeshInstance3DEditor::_create_collision_shape() {
+	int placement_option = shape_placement->get_selected();
+	int shape_type_option = shape_placement->get_selected();
```

Comment:
this is the culprit code I used in both case `shape_placement` while the second one was supposed to be `shape_type` must have been a bad copy paste. :(